pkg/controller/controller.go

```diff
@@ -189,27 +188,30 @@ func NewController(params Params, startController bool) *Controller {
 		ctlr.shareNodes = true
 	}
 
-	if err := ctlr.setupClientsforCluster(params.Config, params.IPAM, ""); err != nil {
+	if err := ctlr.setupClientsforCluster(params.Config, params.IPAM, "", clusterConfig); err != nil {
 		log.Errorf("Failed to Setup Clients: %v", err)
 	}
+	// add the cluster config for local cluster
+	ctlr.multiClusterConfigs.addClusterConfig("", clusterConfig)
 
-	if ctlr.multiClusterConfigs.ClusterConfigs[""].namespaceLabel == "" {
+	if params.NamespaceLabel == "" {
 		if len(params.Namespaces) == 0 {
-			ctlr.multiClusterConfigs.ClusterConfigs[""].namespaces[""] = true
+			clusterConfig.namespaces[""] = true
 			log.Debug("No namespaces provided. Watching all namespaces")
 		} else {
 			for _, ns := range params.Namespaces {
-				ctlr.multiClusterConfigs.ClusterConfigs[""].namespaces[ns] = true
+				clusterConfig.namespaces[ns] = true
 			}
 		}
 	} else {
-		err2 := ctlr.createNamespaceLabeledInformerForCluster(ctlr.multiClusterConfigs.ClusterConfigs[""].namespaceLabel, "")
+		err2 := ctlr.createNamespaceLabeledInformerForCluster(params.NamespaceLabel, "")
 		if err2 != nil {
 			log.Errorf("%v", err2)
-			for _, nsInf := range ctlr.multiClusterConfigs.ClusterInformers[""].nsInformers {
+			informerStore := ctlr.multiClusterConfigs.getInformerStore("")
```

Comment:
What if there is no informer for that cluster, can we add a safety check 

`informerStore`

Reply:
Yeah, Code will be further refactored in upcoming PRs.

Reply:
Ok, Got it~